scripts/helpers/conditionalBuildBabelPlugin.js

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+module.exports = excludedModules => {
+  return {
+    visitor: {
+      ImportDeclaration(path) {
+        let skipWhenComments = [];
+        if (path.node.leadingComments) {
+          skipWhenComments = path.node.leadingComments.filter(c => {
+            return c.value.trim().startsWith("@skipwhen");
+          });
+        }
+
+        if (skipWhenComments.length > 0) {
+          const [, webSDKModuleName, value] = skipWhenComments[0].value.match(
+            "ENV.(.*) === (false|true)"
+          );
+
+          if (excludedModules[webSDKModuleName] === value) {
+            const variableName = path.node.specifiers[0].local.name;
+
+            // Wrap the variable declaration in an IIFE to turn it into an expression
+            path.replaceWithSourceString(
+              `(() => { const ${variableName} = () => {}; })()`
+            );
+          }
+        }
+      },
+      ExportDefaultDeclaration(path) {
+        if (path.node.declaration.type === "ArrayExpression") {
+          path.node.declaration.elements = path.node.declaration.elements.filter(
+            element => {
+              if (element.name) {
+                const variableName = element.name;
+                const componentName = variableName
+                  .replace("create", "")
+                  .toLowerCase();
+                return !Object.keys(excludedModules).includes(
+                  `alloy_${componentName}`
+                );
+              }
+              return true;
+            }
+          );
+        }
+      }
+    }
+  };
+};
```

Comment:
One of the goals of this project is to get Adobe Tags to support this build method. To that end whatever thing we make could be used by other Tags extension developers. There is a lot of specific logic inside of the conditionalBuildBabelPlugin code that only applies to our specific use-case. For example, when I see a directive named @skipwhen it makes me think I could put this in front of anything; however, from this code it looks like you could only use it in front of an import and also if you are using the variable in an array.

I did a bit of research about conditionally building things in javascript. One of the established patterns within javascript libraries is to use an if statement that references process.env. This is especially used when referencing process.env.NODE_ENV. i.e.
```javascript
if (process.env.NODE_ENV === "development") {
  console.log("development mode!");
}
```

I put together a sample to see if this would work with Rollup. I've also noticed that a lot of rollup systems have plugins or ways to accomplish this.  https://github.com/jonsnyder/conditional-compilation-sample

Reply:
Good point @jonsnyder!

@shammowla Please check out Jon's sample and maybe let's schedule a work session to make this PR more reusable.